symphony/lib/core/class.log.php

```diff
@@ -275,6 +275,65 @@ public function pushExceptionToLog($exception, $writeToLog = false, $addbreak =
     }
 
     /**
+     * Given an method name, this function will properly format a message
+     * and pass it down to `pushToLog()`
+     *
+     * @see Log::pushToLog()
+     * @since Symphony 2.7.0
+     * @param string $method
+     *  The name of the deprecated call
+     * @param string $alternative
+     *  The name of the new method to use
+     * @param array $opts (optional)
+     * @param string $opts.message-format
+     *  The sprintf format to apply to $method
+     * @param string $opts.alternative-format
+     *  The sprintf format to apply to $alternative
+     * @param boolean $opts.write-to-log
+     *  If set to true, this message will be immediately written to the log. By default
+     *  this is set to false, which means that it will only be added to the array ready
+     *  for writing
+     * @param boolean $opts.addbreak
+     *  To be used in conjunction with `$opts.write-to-log`, this will add a line break
+     *  before writing this message in the log file. Defaults to true.
+     * @param boolean $opts.append
+     *  If set to true, the given `$message` will be append to the previous log
+     *  message found in the `$_log` array
+     * @param boolean $opts.addtrace
+     *  If set to true, the caller of the function will be added. Defaults to true.
+     * @return boolean|null
+     *  If `$writeToLog` is passed, this function will return boolean, otherwise
+     *  void
+     */
+    public function pushDeprecateWarningToLog($method, $alternative = null, array $opts = array())
```

Comment:
This is nice, only thing I can think of adding would be the removed version (if applicable). eg. `The method `blah` is deprecated and will be removed in v3.0.0. Please use `foo` instead.`